src/Umbraco.Core/Routing/ContentFinderByUrlAndTemplate.cs

```diff
@@ -24,10 +25,10 @@ namespace Umbraco.Cms.Core.Routing;
 ///     </para>
 ///     <para>If successful, then the template of the document request is also assigned.</para>
 /// </remarks>
-public class ContentFinderByUrlAndTemplate : ContentFinderByUrl
+public class ContentFinderByUrlAndTemplate : ContentFinderByUrlNew
```

Comment:
As mentioned by others, this is a breaking change because the signature of the class changes (as are many off the other changes you made)

To only way to do this in a non breaking way, is to create a new ContentFinder with the update logic like we did for `ContentFinderByUrlNew` and obsolete the old one with a reference to the new one. This way we can remove the old one and rename the new one in a full major from the introduction of the new one (in this case as of writing that would be v17)

It is the same reason why `ContentFinderByUrlNew` was added in https://github.com/umbraco/Umbraco-CMS/pull/17033.

When this is done, the article should also be updated to reference the new contentfinder in the v15+ docs

Reply:
Hi @yiangos it seems like this is the only correction that needs to be done to get this merged in.
Do you need some support on making this happen?